packages/roosterjs-editor-api/lib/format/setTextColor.ts

```diff
@@ -1,14 +1,30 @@
 import applyInlineStyle from '../utils/applyInlineStyle';
 import { Editor } from 'roosterjs-editor-core';
+import { ModeIndependentColor } from 'roosterjs-editor-types';
 
 /**
  * Set text color at selection
  * @param editor The editor instance
- * @param color The color string, can be any of the predefined color names (e.g, 'red')
+ * @param color One of two options:
+ * The color string, can be any of the predefined color names (e.g, 'red')
  * or hexadecimal color string (e.g, '#FF0000') or rgb value (e.g, 'rgb(255, 0, 0)') supported by browser.
  * Currently there's no validation to the string, if the passed string is invalid, it won't take affect
+ * Alternatively, you can pass a @typedef ModeIndepenentColor. If in light mode, the lightModeColor property will be used.
+ * If in dark mode, the darkModeColor will be used and the lightModeColor will be set as the ogsc.
  */
-export default function setTextColor(editor: Editor, color: string) {
-    color = color.trim();
-    applyInlineStyle(editor, element => (element.style.color = color));
+export default function setTextColor(editor: Editor, color: string | ModeIndependentColor) {

```

Comment:
This looks very similar to `setBackgroundColor`. Could they be consolidated? Perhaps take a param to set ogsc or ogsb?

Reply:
Fair point, I'd be open to it, but that'd not be in scope for this PR.